### DISCUSSION TITLE
3rd Party Components Integration

### DISCUSSION THREAD

**Discussion by larsontim12** [oqtane/oqtane.framework#906] [General] [answered]
I am trying to find an example of a 3rd party component being implemented into a module without touching the Oqtane base.

I want to be able and add the component on the page and set the properties. I have seen the DNFProjects sample but that doesn't use the component on the index.razor page.

If someone can provide assistance or a sample I would greatly appreciate it!

Thanks -Tim

**Comment by larsontim12** [comment, marked as answer]
Anyone??? Please help....

**Reply by mikecasas** [reply]
What component are you trying to implement?

**Reply by mikecasas** [reply]
Try #693 or #690 

Those are implementing Radzen and Telerik tools.

**Reply by larsontim12** [reply]
I am trying to implement the map component from Syncfusion any component would be helpful.  I tried following the examples from the links above but they jump all over with modifying the core to using themes but I want to make sure it is all encased in the module without touching anything from the core.  

I do appreciate any help you can provide. Examples are the best teaching sources :) I have Telerik controls also.

Thanks - Tim

**Reply by mikecasas** [reply]
I have Radzen running in my external module and I didn't modify any Core code.

1. Create an external module in Oqtane.
2. Then implement ```IClientStartup``` in your external module. Mine is below.

```
 public class ModuleStartup : IClientStartup
    {
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddScoped<Radzen.DialogService>();
        }       
    }
```

3. Include any JS needed in the wwwroot/Modules of your external module's Server project. For my project, I have the ```Radzen.Blazor.js``` file in the folder.

4. Make sure in the Package project of your external module, you "push/copy" all the assets to the oqtane bin folder. Below is the additional code needed for the third party assets

```
REM Copy the Radzen DLL to the Bin folder (and any dll dependencies)
XCOPY "..\Client\bin\Debug\netstandard2.1\Radzen.Blazor.dll" "..\..\oqtane.framework\Oqtane.Server\bin\Debug\netcoreapp3.1\" /Y
XCOPY "..\Client\bin\Debug\netstandard2.1\System.Linq.Dynamic.Core.dll" "..\..\oqtane.framework\Oqtane.Server\bin\Debug\netcoreapp3.1\" /Y

```

I got the DLL from downloading the Radzen nuget package and extracting it.


5. Then in my ```Index.razor``` I can use.

```
 <RadzenScheduler @ref="scheduler" style="height: 768px;" TItem="Appointment" Data="@appointments" StartProperty="Start" EndProperty="End"
                            TextProperty="Text" SelectedIndex="2">
            <RadzenDayView />
            <RadzenWeekView />
            <RadzenMonthView />
        </RadzenScheduler>
```


I think this is all you need.

**Reply by larsontim12** [reply]
Thank you very much! I will give it a shot.

The issue I see with copying the dll is what if you have other modules that are using different versions. I am sure that is something I can figure out later but seems like it can be an issue with modules from different people and not being able to upgrade at the same time.

Again thank you for the help! -Tim

**Reply by mikecasas** [reply]
That I have not tested.

**Reply by ADefWebserver** [reply]
I posted a Blog on this: [Using Radzen In Oqtane Modules](https://blazorhelpwebsite.com/ViewBlogPost/46)

**Comment by dosni** [comment]
I tried following instructions from blazorhelp to use radzen in oqtane.  It works but When I added radzen button, the styling lost.  Is there any steps that I missed ?

![WhatsApp Image 2023-09-23 at 16 33 41](https://github.com/oqtane/oqtane.framework/assets/87328342/4f2be387-d2c3-49d3-8a14-cdc15a98b618)

**Comment by dosni** [comment]
I solved it already using recommendation from this discussion , 
[2024](https://github.com/oqtane/oqtane.framework/discussions/2024)
